packages/compiler/src/experimental/typekit/kits/record.ts

```diff
@@ -43,7 +44,12 @@ defineKit<TypekitExtension>({
     },
     getElementType(type) {
       if (!this.record.is(type)) {
-        throw new Error("Type is not a record.");
+        reportTypekitDiagnostic(this.program, {
```

Comment:
reporting diagnostic from a getter is a bad pattern, it carries the error to the user potentially multiple times without expoissting that to the library author.

Any getter that report diagnmostic should return the error tuple

Reply:
in general the same applies to probably all typekits

Reply:
i think it might make more sense to also throw if invalid inputs are used